repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -73,19 +63,29 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
     }
   }
 
-  def handle(ctx: ChannelHandlerContext, msg: BaseProtocol.ReplJobRequest): Unit = {
-    Future {
-      jobFutures(msg.id) = session.execute(msg.code).result
-    }
+  def handle(ctx: ChannelHandlerContext, msg: BaseProtocol.ReplJobRequest): Int = synchronized {
```

Comment:
I'm not sure here `synchronized` is required? From my understanding it is not necessary, since `session.execute` has already been synchronized.